multiversx_sdk_core/transaction_factories/smart_contract_transactions_factory.py

```diff
@@ -57,24 +59,83 @@ def create_transaction_for_deploy(self,
+                                       token_transfers: List[TokenTransfer] = []) -> Transaction:
+        number_of_tokens = len(token_transfers)
+
+        if native_transfer_amount and number_of_tokens:
+            raise BadUsageError("Can't send both native token and ESDT/NFT tokens")
+
+        token_computer = TokenComputer()
+        transfer_args: List[str] = []
+
+        if len(token_transfers) == 1:
+            transfer = token_transfers[0]
+
+            if token_computer.is_fungible(transfer.token):
+                transfer_args = self._build_args_for_esdt_transfer(transfer)
+                transfer_args.append(arg_to_string(function))
+                transfer_args.extend(args_to_strings(arguments))
+            else:
+                transfer_args = self._build_args_for_single_esdt_nft_transfer(transfer)
+                transfer_args.append(contract.hex())
+                contract = sender
+                transfer_args.append(arg_to_string(function))
+                transfer_args.extend(args_to_strings(arguments))
+        elif len(token_transfers) > 1:
+            transfer_args = self._build_args_for_multi_esdt_nft_transfer(contract.hex(), token_transfers)
+            contract = sender
+            transfer_args.append(arg_to_string(function))
+            transfer_args.extend(args_to_strings(arguments))
+        else:
+            transfer_args.append(function)
+            transfer_args.extend(args_to_strings(arguments))
 
-        intent = TransactionBuilder(
+        transaction = TransactionBuilder(
             config=self.config,
             sender=sender,
-            receiver=contract_address,
-            data_parts=parts,
+            receiver=contract,
+            data_parts=transfer_args,
             gas_limit=gas_limit,
-            add_data_movement_gas=False
+            add_data_movement_gas=False,
+            amount=native_transfer_amount
         ).build()
 
-        return intent
+        return transaction
+
+    def _build_args_for_esdt_transfer(self, transfer: TokenTransfer) -> List[str]:
+        args: List[str] = ["ESDTTransfer"]
+        args.extend(args_to_strings([transfer.token.identifier, transfer.amount]))
+        return args
+
+    def _build_args_for_single_esdt_nft_transfer(self, transfer: TokenTransfer) -> List[str]:
+        args: List[str] = ["ESDTNFTTransfer"]
+        token = transfer.token
+        identifier = self._ensure_identifier_has_correct_structure(token.identifier)
+        args.extend(args_to_strings([identifier, token.nonce, transfer.amount]))
+        return args
+
+    def _build_args_for_multi_esdt_nft_transfer(self, contract: str, transfers: List[TokenTransfer]) -> List[str]:
+        args: List[str] = ["MultiESDTNFTTransfer", contract, arg_to_string(len(transfers))]
+
+        for transfer in transfers:
+            identifier = self._ensure_identifier_has_correct_structure(transfer.token.identifier)
```

Comment:
Here and above, `TokenComputer.extract_identifier_from_extended_identifier` can be called directly. Also, maybe the token computer can be received as a dependency within the factory's constructor (as an interface). Similar to how was done for transaction serializer & address computer. 

Can be changed in the next PR.